cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -781,4 +790,102 @@ private void writeSystemMetadata(Id.Artifact artifactId, ArtifactInfo artifactIn
     ArtifactSystemMetadataWriter writer = new ArtifactSystemMetadataWriter(metadataStore, artifactId, artifactInfo);
     writer.write();
   }
+
+  /**
+   * Given plugin artifact, find the parent artifact of plugin, after creating classloader with parent artifact
+   * and plugin, we invoke the plugin method in the plugin identified by type and name and return the response.
+   * @param namespace namespace
+   * @param artifactId plugin artifact id
+   * @param pluginType type of the plugin
+   * @param pluginName name of the plugin
+   * @param methodName name of the method
+   * @param parameters request body which corresponds to method parameter
+   * @return Serialized string response from method invocation
+   * @throws ArtifactNotFoundException
+   * @throws IOException
+   * @throws IllegalArgumentException
+   * @throws BadRequestException
+   * @throws ClassNotFoundException
+   * @throws InvocationTargetException
+   */
+  public String pickParentArtifactAndInvokePluginEndpoint(Id.Namespace namespace,
```

Comment:
ok i think i get it, this is the namespace of the caller, since the namespace of the artifact is either that namespace or the system namespace.  The javadoc should clarify.